owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoteShare.kt

```diff
@@ -0,0 +1,241 @@
+ *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+ *
+ *   The above copyright notice and this permission notice shall be included in
+ *   all copies or substantial portions of the Software.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ *
+ */
+
+package com.owncloud.android.lib.resources.shares
+
+import android.os.Parcel
+import android.os.Parcelable
+import com.owncloud.android.lib.common.utils.Log_OC
+import com.owncloud.android.lib.resources.files.FileUtils
+import java.io.Serializable
+
+/**
+ * Contains the data of a Share from the Share API
+ *
+ * @author masensio
+ * @author David A. Velasco
+ * @author David González Verdugo
+ */
+class RemoteShare : Parcelable, Serializable {
+
+    /// Getters and Setters
+    var fileSource: Long = 0
+    var itemSource: Long = 0
+    var shareType: ShareType? = null
+    private var mShareWith: String? = null
+    private var mPath: String? = null
+    var permissions: Int = 0
+    var sharedDate: Long = 0
+    var expirationDate: Long = 0
+    private var mToken: String? = null
+    private var mSharedWithDisplayName: String? = null
+    var sharedWithAdditionalInfo: String? = null
+    private var mName: String? = null
+    var isFolder: Boolean = false
+    var userId: Long = 0
+    var remoteId: Long = 0
+        private set
+    private var mShareLink: String? = null
+
+    var shareWith: String?
+        get() = mShareWith
+        set(shareWith) {
+            this.mShareWith = shareWith ?: ""
+        }
+
+    var path: String?
+        get() = mPath
+        set(path) {
+            this.mPath = path ?: ""
+        }
+
+    var token: String?
+        get() = mToken
+        set(token) {
+            this.mToken = token ?: ""
+        }
+
+    var sharedWithDisplayName: String?
+        get() = mSharedWithDisplayName
+        set(sharedWithDisplayName) {
+            this.mSharedWithDisplayName = sharedWithDisplayName ?: ""
+        }
+
+    var name: String?
+        get() = mName
+        set(name) {
+            mName = name ?: ""
+        }
+
+    var shareLink: String?
+        get() = this.mShareLink
+        set(shareLink) {
+            this.mShareLink = shareLink ?: ""
+        }
+
+    val isPasswordProtected: Boolean
+        get() = ShareType.PUBLIC_LINK == shareType && mShareWith!!.length > 0
```

Comment:
ShareType.PUBLIC_LINK == shareType && shareWith?.isNotEmpty == true

BTW, why are you declaring shareWith as nullable String but you are asserting here it won't be

Reply:
It will not be nullable any more